### PR TITLE
Fix 'test' make target for java s2i

### PR DESCRIPTION
**Pull request by tmckayus**
The test target runs s2i tests on the image, verifying that
the s2i scripts (assemble, run, and usage) work.

### CODE REVIEW

**Comment by tmckayus** [comment]
This is a partial fix for #49

File: java/java/test/run [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+# The 'run' performs a simple test that verifies the S2I image.
+# The main focus here is to exercise the S2I scripts.
+#
+# For more information see the documentation:
+# https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md
+#
+# IMAGE_NAME specifies a name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+IMAGE_NAME=${IMAGE_NAME-s2i-test-candidate}
```
Comment (elmiko) [inline java/java/test/run]: should this be `IMAGE_NAME=${IMAGE_NAME}-s2i-test-candidate` ?

File: java/java/test/run [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+# Determining system utility executables (darwin compatibility check)
+READLINK_EXEC="readlink"
+MKTEMP_EXEC="mktemp"
```
Comment (elmiko) [inline java/java/test/run]: maybe we should use the `$(which readlink)` format here?

File: java/java/test/run (hunk 1 above)
Comment (tmckayus) [inline java/java/test/run]: Yeah, that is weird.  Same in the py case.  Copied from the s2i sample project test.

File: java/java/test/run (hunk 2 above)
Comment (tmckayus) [inline java/java/test/run]: didn't originate with us, but yeah, I have no problem with $(which xxxx).  Will change.

**Pull request merged**